repo/content/content_manager.go

```diff
@@ -439,6 +436,18 @@ func (bm *Manager) Flush(ctx context.Context) error {
 		bm.cond.Broadcast()
 	}()
 
+	// see if we have any packs that have failed previously
```

Comment:
Is this changing the existing behavior / semantics in terms of:
1. not starting a new pack if there failed ones that need to be retried;
2. the frequency of the retries?

Perhaps I'm missing something.

Reply:
It does change the behavior a bit:

- previously all contents were added to the index AFTER they finished writing and it could be the index blob that's currently being built or the next one

- with this change contents are pre-destined for the current index blob and if the packs fails to write, the index won't be written until the pack is retried

it should not change much in terms of timings or performance since index blobs are only flushed at the end of the session or every N minutes.

Reply:
I understand the change regarding the semantics for timed flushed.

The question was related to the change for retrying failed packs.
- Before a content write would be rejected (failed) if there were failed packs and the retry failed. Essentially, this provided back pressure to avoid accumulating an unbounded number of writes. That was the original motivation. Now it seems that writes would be accumulated until the next timed or explicit flush.
- And secondary, the frequency of retries is much lower, which could be desirable to avoid retrying too frequently, but at the same time if the retry period is very large, that may be undesirable.

Again, maybe I'm missing something. I'll take another deeper pass at this portion.

Reply:
For completeness (from a separate convo):

IIUC, `WriteContent()` calls `addToPackUnlocked` for new (not-already-existing) content.
- Previously, the retry was in `addToPackUnlocked` before adding the content to the current pack & index. It was performed for every write of new content.
- Now the retry happens conditionally in `addToPackUnlocked` only after `bm.flushPackIndexesAfter`. So, with defaults, the worst case would be 10min. Probably 5 min on average.

Reply:
Pushed one more commit that will explicitly retry previously-failed packs before WriteContent() to restore the previous behavior.